storage-proofs/porep/src/stacked/vanilla/memory_handling.rs

```diff
@@ -77,7 +148,8 @@ impl<T: FromByteSlice> CacheReader<T> {
     // This allows splitting the reset in order to avoid a pause.
     pub fn reset(&self) -> Result<()> {
         self.start_reset()?;
-        self.finish_reset()
+        self.finish_reset()?;
+        Ok(())
```

Comment:
Just curious why you split this?

Reply:
It lets the first remapping happen while the previous layer is written to disk. In theory, it can happen sooner (while the previous layer is still being finished). That requires a bit more bookkeeping to get right, though — so the current implementation is enough for now.

Reply:
I just meant the return of the finish result, compared to swallowing the result and explicitly returning Ok (the before/after seem comparable), just more verbose now.  Maybe I'm missing something.

Reply:
Oh, good point. I didn't actually do that on purpose. It's probably a casualty of the interactive-rebase/fix-errors slaughterhouse. I'll change it back.